agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -16,12 +16,17 @@ class ServicePodWorker
     attr_reader :node, :prev_state, :service_pod
     attr_accessor :service_pod, :container_state_changed
 
+    DIE_EVENTS = ['die', 'kill'].freeze
```

Comment:
I think only `die` should be handled. `docker kill/stop xyz` emits also `die` event so AFAIK it should be safe to only handle `die` event. Handling both makes the restart logic to be executed twice in case of `docker kill/stop xyz` as that emits both `kill`and `die`.

`docker kill nginx-1` results in:
```
kontena-agent | I, [2017-08-15T10:15:51.164164 #1]  INFO -- Kontena::Workers::ServicePodWorker: restarting nginx-1 because it has stopped
kontena-agent | D, [2017-08-15T10:15:51.164448 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of nginx-1: running
kontena-agent | D, [2017-08-15T10:15:51.177731 #1] DEBUG -- Kontena::RpcClient: waited 0.0s of 30.0s until: request /containers/event has response wth id=2044946063 yielded Array
kontena-agent | D, [2017-08-15T10:15:51.187778 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state is in-sync: running
kontena-agent | D, [2017-08-15T10:15:51.247931 #1] DEBUG -- Kontena::RpcClient: waited 0.1s of 30.0s until: request /containers/save has response wth id=63854677 yielded Array
kontena-agent | I, [2017-08-15T10:15:51.381047 #1]  INFO -- Kontena::Workers::ServicePodWorker: restarting nginx-1 because it has stopped (delay: 1s)
kontena-agent | D, [2017-08-15T10:15:51.393973 #1] DEBUG -- Kontena::RpcClient: waited 0.0s of 30.0s until: request /containers/event has response wth id=382750331 yielded Array
kontena-agent | D, [2017-08-15T10:15:52.382805 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of nginx-1: running
kontena-agent | D, [2017-08-15T10:15:53.866767 #1] DEBUG -- Kontena::RpcClient: waited 0.0s of 30.0s until: request /containers/event has response wth id=1151646354 yielded Array
```